### DISCUSSION TITLE
How to modify guest middleware included in laravel framework?

### DISCUSSION THREAD

**Discussion by VladRafli** [laravel/framework#51911] [Q&A]
Hello everyone,

I'm currently working on creating website using **Laravel 11** with scaffolding using **Breeze**. I had some issues related to middleware that included in laravel and used on route generated by Breeze.

I have my routes.php looks like this:

```php
Route::middleware('guest')->get(function () {
        if (!Auth::check()) {
            return redirect()->intended(route('auth.login'));
        }

        if (Auth::user()->hasRole([UserRolesEnum::ADMIN, UserRolesEnum::SUPER_ADMIN])) {
            return redirect(route('admin.dashboard'));
        }

        if (Auth::user()->hasRole(UserRolesEnum::USER)) {
            return redirect(route('user.dashboard'));
        }
});

Route::middleware('guest')->prefix('auth')->group(function () {
    Route::get('login', [AuthenticatedSessionController::class, 'create'])->name('auth.login');
    // other routes...
});
```

Had some changes on AuthenticatedSessionController::class only to check if user's role for redirecting to spesific route

```php
class AuthenticatedSessionController extends Controller
{
     /**
     * Handle an incoming authentication request.
     */
    public function store(LoginRequest $request): RedirectResponse
    {
        $request->validate([
            'email' => 'required|email',
            'password' => 'required|string',
        ]);

        $request->authenticate();

        $request->session()->regenerate();

        if (Auth::user()->hasRole([UserRolesEnum::ADMIN, UserRolesEnum::SUPER_ADMIN])) {
            return redirect()->intended(route('admin.dashboard', absolute: false));
        }

        if (Auth::user()->hasRole([UserRolesEnum::USER])) {
            return redirect()->intended(route('user.dashboard', absolute: false));
        }

        return redirect()->intended(route('auth.login'));
    }
}
```

After successfully attempting to login and had some error because I haven't implemented route for 'user.dashboard' and 'admin.dashboard' then refresh the page, it redirect me to path '/' which I don't remember to redirect there when user authenticated. When I tried to access path 'auth/login' it redirects me to '/' because I assume the user already logged in and 'guest' middleware tries to redirect me to default path when user is authenticated.

So I tried to retrace my step when creating the login feature, and found that when removing 'guest' middleware make the feature  work as intended and not redirecting me to '/' again.

```php
Route::get(function () {
        if (!Auth::check()) {
            return redirect()->intended(route('auth.login'));
        }

        if (Auth::user()->hasRole([UserRolesEnum::ADMIN, UserRolesEnum::SUPER_ADMIN])) {
            return redirect(route('admin.dashboard'));
        }

        if (Auth::user()->hasRole(UserRolesEnum::USER)) {
            return redirect(route('user.dashboard'));
        }
});

Route::prefix('auth')->group(function () {
    Route::get('login', [AuthenticatedSessionController::class, 'create'])->name('auth.login');
    // other routes...
});
```

And I remember how to change this behaviour when using Laravel 10 before to change route to home page or something inside middleware RedirectIfAuthenticated.php which is part of 'guest' middleware, but in Laravel 11, this file is moved inside the framework itself which it surely cannot be modified directly.

I haven't found a reference how to solve this, so might be someone might had found its solution would be appreciated sharing it here.

Thank you.

**Comment by albetnov** [comment]
You can configure it at `app/bootstrap.php` in the `withMiddleware` method:
```php
->withMiddleware(function (Middleware $middleware) {
        $middleware->redirectUsersTo(function(Request $request) {
            return $request->user()->isAdmin() ? '/admin' : '/home';
        });
    })
```

**Reply by VladRafli** [reply]
Is there any link that I can use as future reference inside Laravel Docs?

**Reply by albetnov** [reply]
You can look at the API Documentation provided by Laravel. 
For this case: https://laravel.com/api/11.x/Illuminate/Foundation/Configuration/Middleware.html

**Reply by VladRafli** [reply]
Okay, but what about in "/docs" page? I think this middleware settings isn't mentioned there? Because I always tried find things inside docs as I'm myself not very familiar with Laravel.

**Reply by VladRafli** [reply]
I only found this one resource link that help me find that particular middleware configuration.

https://laraveldaily.com/post/change-redirect-login-register-laravel-breeze
